### PR TITLE
Rh aleem signup

### PR DESCRIPTION
**Pull request by aleemcahn**
As function based views were already used in the code so, I also used FBV, but I think we should use class based views for future inheritance and code. 
Also in the original ReportHub the signup process is long, I covered the basic part of authentication, but we can further add things in the signup registration form.

### CODE REVIEW

File: rh/decorators.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.shortcuts import redirect
+
+def unauthenticated_user(view_func):
```
Comment (mishaker) [inline rh/decorators.py:3]: nice wrapper !

File: rh/views.py [hunk 2]
```diff
@@ -1,18 +1,74 @@
-from email import message
-from django.shortcuts import render
+from functools import cache
+import pandas as pd
+import datetime
+import sqlite3
+from multiprocessing import context
+from tkinter import E
```
Comment (mishaker) [inline rh/views.py:6]: I imagine you forgot to remove this one ?

File: rh/views.py [hunk 3]
```diff
@@ -1,18 +1,74 @@
...
-import sqlite3
 
+def logout_view(request):
+    """User Logout View"""
+    messages.info(request, f'{request.user} logged out.')
+    logout(request)
+    return redirect("/login")
+
+@cache_control(no_store=True)
+@login_required
```
Comment (mishaker) [inline rh/views.py:71]: index page is public. doesn't require login.

File: templates/registration/signup.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    </div>
+</div>
+
+{% comment %} <script>
+    $('#id_password2').keyup(function(){
+        var pass1 = $('#id_password1');
+        var pass2 = $('#id_password2');
+        if(pass1.val() != pass2.val()){
+            debugger
+            pass1.attr({style:'box-shadow: 0 0 5px rgb(225 84 84)'});
```
Comment (mishaker) [inline templates/registration/signup.html:61]: if we can extract these styles into CSS file and use django compressor workflow that would be great.

File: templates/registration/signup.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        </div>
+    </div>
+</div>
+
+{% comment %} <script>
+    $('#id_password2').keyup(function(){
+        var pass1 = $('#id_password1');
+        var pass2 = $('#id_password2');
+        if(pass1.val() != pass2.val()){
+            debugger
```
Comment (mishaker) [inline templates/registration/signup.html:60]: I imagine the debugger won't be necessary in production ?

**Review by mishaker — CHANGES_REQUESTED** [review]
Overall great PR ! thanks ! 
I only have few minor comments.

File: rh/views.py (hunk 2 above)
Comment (aleemcahn) [reply rh/views.py:6]: Yes I forgot removing these xD

**Review by mishaker — APPROVED** [review]
LGTM 👍  I let you merge it once you are done ! thanks :)

**Pull request merged**